Data/Row/Dictionaries.hs

```diff
@@ -101,6 +105,10 @@ mapExtendSwap = UNSAFE.unsafeCoerce $ Dict @Unconstrained
 apExtendSwap :: forall k ℓ (τ :: k) r (f :: k -> *) fs. Dict (Extend ℓ (f τ) (Ap fs r) ≈ Ap (Extend ℓ f fs) (Extend ℓ τ r))
 apExtendSwap = UNSAFE.unsafeCoerce $ Dict @Unconstrained
 
+-- | Proof that the 'ApSingle' type family preserves labels and their ordering.
+apSingleExtendSwap :: forall k ℓ (τ :: k) r (f :: k -> *). Dict (Extend ℓ (f τ) (ApSingle r τ) ≈ ApSingle (Extend ℓ f r) τ)
+apSingleExtendSwap = UNSAFE.unsafeCoerce $ Dict @Unconstrained
```

Comment:
I'm not sure you need the `k` argument here, and having it makes type application a little more annoying.

Reply:
The `k` isn't necessary in the type signature, but my understanding is that it will still be necessary in the type application (so in my opinion leaving it in the type signature makes things more clear and explicit). The behaviour is described in https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/glasgow_exts.html#ordering-of-specified-variables. Specifically, the second bullet point describes how kind variables are re-ordered to the front.

I've verified this experimentally by removing `k` from the type signature. Even with `k` removed, the type application still requires `@_` before the label is specified, and ghci includes it in the type.

```
*Data.Row.Dictionaries> :t +v apSingleExtendSwap
apSingleExtendSwap
  :: forall k (ℓ :: Symbol) (τ :: k) (r :: Row (k -> *))
            (f :: k -> *).
     Dict (Extend ℓ (f τ) (ApSingle r τ) ≈ ApSingle (Extend ℓ f r) τ)
```

As it turns out, the only reason I figured this out in the first place is because I couldn't figure out why metamorph seemed to need an extra `@_` whenever applying types to it! So perhaps it's worth considering adding the `k` to other type signatures so it's more explicit; or removing `k` from them for consistency (I was after all trying to be consistent with `apExtendSwap` in the first place).